src/viz/RobotModel.js

```diff
@@ -100,16 +104,40 @@ class RobotModel extends URDFLoader {
   }
 
   destroy() {
+    const robotModelParent = this.robotModel && this.robotModel.parent;
+    if (robotModelParent) {
+      // TODO: proper scene clear for all viz types
+      // just parent.remove is not enough
+      Object.values(this.robotModel.links).map(linkObj => {
+        if (linkObj.parent) {
+          linkObj.parent.remove(linkObj);
+        }
+      });
+      robotModelParent.remove(this.robotModel);
```

Comment:
The model itself need not be removed. Irrespective of whether or not it's in the scene the destroy function should work